backend/layers/processing/schema_migration.py

```diff
@@ -263,6 +269,10 @@ def log_errors_and_cleanup(self, collection_version_id: str) -> list:
         self.s3_provider.delete_files(self.artifact_bucket, object_keys_to_delete)
         if errors:
             self._store_sfn_response("report/errors", collection_version_id, errors)
+            # clean up artifacts for any now-orphaned, rolled back datasets
+            if rolled_back_datasets:
+                # TODO: replace with async batch job to delete orphaned dataset version DB rows + artifacts
```

Comment:
A batch job seems heavy handed to make API calls. I think you should do it as a lambda, or as a subprocess or thread of this batch job.

Reply:
I'm fine with the idea of using a lambda (or another alternative) instead. Updated the comment to be agnostic to infra type